test/test.js

```diff
@@ -890,340 +919,443 @@ describe('S3rver Tests', function () {
+    const url = s3Client.getSignedUrl('getObject', params);
+    const s3rver = new S3rver({
+      port: 4569,
+      hostname: 'localhost',
+      silent: true,
+      cors: fs.readFileSync('./test/resources/cors_test1.xml')
+    }).run((err) => {
+      if (err) return done(err);
+
+      request({ url, headers: { origin, range: 'bytes=0-100' } }, function (err, response, body) {
+        s3rver.close(() => {
+          if (err) return done(err);
+
+          response.statusCode.should.equal(206);
+          response.headers.should.have.property('access-control-expose-headers', 'Accept-Ranges, Content-Range');
+          done();
+        });
+      });
     });
   });
 });
 
 describe('S3rver Tests with Static Web Hosting', function () {
   let s3Client;
   let s3rver;
-  beforeEach(function (done) {
+
+  beforeEach('Reset site bucket', resetTmpDir);
+  beforeEach('Start s3rver', function (done) {
     s3rver = new S3rver({
       port: 5694,
       hostname: 'localhost',
       silent: true,
       indexDocument: 'index.html',
       errorDocument: '',
-      directory: '/tmp/s3rver_test_directory'
+      directory: tmpDir
     }).run(function (err, hostname, port, directory) {
-        if (err) {
-          return done('Error starting server', err);
-        }
-        const config = {
-          accessKeyId: '123',
-          secretAccessKey: 'abc',
-          endpoint: util.format('%s:%d', hostname, port),
-          sslEnabled: false,
-          s3ForcePathStyle: true
-        };
-        AWS.config.update(config);
-        s3Client = new AWS.S3();
-        s3Client.endpoint = new AWS.Endpoint(config.endpoint);
-        /**
-         * Remove if exists and recreate the temporary directory
-         */
-        fs.remove(directory, function (err) {
-          if (err) {
-            return done(err);
-          }
-          fs.mkdirs(directory, done);
-        });
-      });
-  });
-
-  afterEach(function (done) {
-    s3rver.close(done);
-  });
-
-  it('should create a site bucket', function (done) {
-    s3Client.createBucket({Bucket: 'site'}, function (err) {
       if (err) {
         return done(err);
       }
+      s3Client = new AWS.S3({
+        accessKeyId: '123',
+        secretAccessKey: 'abc',
+        endpoint: util.format('http://%s:%d', hostname, port),
+        sslEnabled: false,
+        s3ForcePathStyle: true
+      });
       done();
     });
   });
 
+  afterEach(function (done) {
+    s3rver.close(done);
+  });
+
   it('should upload a html page to / path', function (done) {
-    s3Client.createBucket({Bucket: 'site'}, function () {
-      const params = {Bucket: 'site', Key: 'index.html', Body: '<html><body>Hello</body></html>'};
+    const bucket = 'site';
+    s3Client.createBucket({Bucket: bucket}, function () {
+      const params = {Bucket: bucket, Key: 'index.html', Body: '<html><body>Hello</body></html>'};
       s3Client.putObject(params, function (err, data) {
-        if (err) return done(err);
```

Comment:
Thanks for adhering to the linting rules as they are... they're not pretty but updating them is for a future PR.